addon/list-utils.js

```diff
@@ -142,7 +142,12 @@ export function getItemsFromEmberData ({value, modelDef, data, bunsenId, store,
   const modelType = modelDef.modelType || 'resources'
   const {labelAttribute, queryForCurrentValue, valueAttribute} = modelDef
   const valueAsId = get(value, bunsenId)
-  const actuallyFindCurrentValue = keepCurrentValue && queryForCurrentValue && valueAsId !== undefined
+  let arrayValues
+  if (typeOf(valueAsId) === 'array' && queryForCurrentValue) {
+    arrayValues = valueAsId.asMutable()
+    arrayValues = RSVP.Promise.all(arrayValues.map((id) => store.findRecord(modelType, id)))
```

Comment:
No need to call `asMutable` if you're not going to mutate it. You might consider just inlining it with the `map`

Reply:
I added a comment explaining why I'm calling `asMutable()`